src/text_modifications.rs

```diff
@@ -219,3 +219,22 @@ mod max_consecutive_chars {
         );
     }
 }
+
+//=====================================
+// General-purpose formatting utilities
+//=====================================
+
+/// `Repeated(content, count` formats as `content` repeated `count` times.
+#[derive(Debug)]
+pub(crate) struct Repeated<T>(pub T, pub usize);
+
+impl<T: fmt::Display> fmt::Display for Repeated<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let Repeated(content, count) = self;
+
+        for _ in 0..*count {
+            write!(f, "{content}")?;
```

Comment:
Let's go with something like `T::fmt(f)` here to avoid the redundant creation of format arguments.

Reply:
I incorporated this change and just rebased now that #103 is merged. Let me know if you see anything else I can improve! :)